raiden/transfer/views.py

```diff
@@ -0,0 +1,273 @@
+    )
+
+    result = []
+    for channel_state in token_network.channelidentifiers_to_channels.values():
+        if channel_state.partner_state.balance_proof:
+            result.append(channel_state)
+
+    return result
+
+
+def get_channelstate_open(node_state, payment_network_id, token_network_id):
+    """Return the state of open channels in a token network."""
+    token_network = get_token_network(
+        node_state,
+        payment_network_id,
+        token_network_id,
+    )
+
+    result = []
+    for channel_state in token_network.channelidentifiers_to_channels.values():
+        if channel.get_status(channel_state) == CHANNEL_STATE_OPENED:
+            result.append(channel_state)
+
+    return result
+
+
+def get_channelstate_not_settled(node_state, payment_network_id, token_network_id):
+    """Return the state of open channels in a token network."""
+    token_network = get_token_network(
+        node_state,
+        payment_network_id,
+        token_network_id,
+    )
+
+    result = []
+    for channel_state in token_network.channelidentifiers_to_channels.values():
+        if channel.get_status(channel_state) == CHANNEL_STATE_SETTLED:
+            result.append(channel_state)
+
+    return result
+
+
+def get_channelstate_by_tokenaddress(node_state, payment_network_id, token_network_id, channel_id):
+    token_network = get_token_network(
+        node_state,
+        payment_network_id,
+        token_network_id,
+    )
+
+    channel_state = None
+    if token_network:
+        channel_state = token_network.channelidentifiers_to_channels.get(channel_id)
+
+    return channel_state
+
+
+def get_transfer_role(node_state, hashlock):
+    transfer_task = node_state.payment_mapping.hashlocks_to_task.get(hashlock)
+
+    result = None
+    if isinstance(transfer_task, PaymentMappingState.InitiatorTask):
+        result = 'initiator'
+    elif isinstance(transfer_task, PaymentMappingState.MediatorTask):
+        result = 'mediator'
+    elif isinstance(transfer_task, PaymentMappingState.TargetTask):
+        result = 'target'
+
+    return result
+
+
+def list_channelstate_for_tokennetwork(node_state, payment_network_id, token_network_id):
+    token_network = get_token_network(
+        node_state,
+        payment_network_id,
+        token_network_id,
+    )
+
+    if token_network:
+        result = token_network.partneraddresses_to_channels.values()
+    else:
+        result = []
+
+    return result
+
+
+def list_channelstate_for_partner(node_state, payment_network_id, partner_address):
+    payment_network = node_state.identifiers_to_paymentnetworks.get(payment_network_id)
+
+    result = []
+    if payment_network is not None:
+
+        for token_network in payment_network.tokenaddresses_to_tokennetworks.values():
+            channel_state = token_network.partneraddresses_to_channels.get(partner_address)
+            if channel_state:
+                # TODO: Either enforce immutability or make a copy
```

Comment:
Does that need to be fixed before merging?

Reply:
this wont be a problem as long as the returned object is not mutate, which is not done at the moment